### PR TITLE
Migrate ccache to sccache

### PR DESCRIPTION
**Pull request by bddppq**

### CODE REVIEW

**Comment by bddppq** [comment]
@pietern

File: install.sh [hunk 1]
```diff
@@ -11,15 +11,24 @@ mkdir -p "$BUILD_DIR"
 
 _pip_install() {
     if [[ -n "$CI" ]]; then
-        ccache -z
+        if [[ -n "${SCCACHE_BUCKET}" ]]; then
+            # Does sccache have flag for clear stats?
```
Comment (pietern) [inline install.sh]: It starts a daemon inside the container and the stats are per daemon run, which in this case is per build.

File: jenkins/setup.sh [hunk 2]
```diff
@@ -6,20 +6,41 @@ export TOP_DIR=$(dirname $(dirname $(readlink -e "${BASH_SOURCE[0]}")))
...
-        export PATH="$TOP_DIR/ccache:$PATH"
+    if [[ -n "${SCCACHE_BUCKET}" ]]; then
+        if hash sccache 2>/dev/null; then
+            echo "SCCACHE_BUCKET is set but sccache executable is not found"
+            exit 1
+        fi
+        SCCACHE_BIN_DIR="$TOP_DIR/sccache"
+        mkdir -p "$SCCACHE_BIN_DIR"
+        for compiler in "${compilers[@]}"; do
+            ln -sf "$(which sccache)" "$SCCACHE_BIN_DIR/$compiler"
```
Comment (pietern) [inline jenkins/setup.sh]: The symlink didn't work in my experience. See caffe2/.jenkins/build.sh for my solution.

Comment (bddppq) [reply jenkins/setup.sh]: @pietern I saw that and was wondering why?

**Pull request merged**